lib/util/git/index.ts

```diff
@@ -42,6 +42,90 @@ export { setPrivateKey } from './private-key';
 // TODO: fix upstream types https://github.com/steveukx/git-js/issues/704
 const ResetMode = (simpleGit.default as any).ResetMode as typeof _ResetMode;
 
+// A generic wrapper for simpleGit.* calls to make them more fault-tolerant
+async function gitRetry(
+  retryCount: number,
+  delaySeconds: number,
+  delayFactor: number,
+  gitFuncName: string,
+  gitFunc: (...args) => Promise<any>,
+  args: any[] | undefined
+): Promise<any> {
```

Comment:
Needs more type safety.

@JamieMagee can you help?

Reply:
This is not needed anymore